### PR TITLE
Ccpansys/solutions e2e testing

### PR DESCRIPTION
**Pull request by ccpansys**
This PR adds a workflow file that will trigger the [solution templates e2e tests](https://github.com/Solution-Applications/solution-templates-end-to-end-testing) on each PR that modifies one of the Solution templates.

### CODE REVIEW

File: .github/workflows/solutions_ci.yml
Comment (jacobevansgit) [inline .github/workflows/solutions_ci.yml:27]: I am wondering if all this logic is needed. Instead could this run be trigged by this?

```yaml
on:
  pull_request:
    branches: [ main ]
    paths:
      - 'your-directory-path/**
```

Usually your logic is needed since it operates in a single `ci.yaml` file, but here you are separated. So maybe could be simplified.

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [inline .github/workflows/solutions_ci.yml:38]: I suggest using the GH CLI instead of calling the REST API directly: https://cli.github.com/manual/gh_workflow_run

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [inline .github/workflows/solutions_ci.yml:41]: To handle authentication with the GH CLI, you can do something like this: https://github.com/Solution-Applications/actions/pull/115/files#diff-cf5ec9da321459d0d1b98c7aeb3fb8e2836ff2a3da032b472c5f5e2720cb3223R84-R85

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [inline .github/workflows/solutions_ci.yml:40]: Just a general suggestion, but I always prefer to use the long-form command-line switches - `--headers` here, I think. It makes it a lot easier to see/understand the meaning of the code instead of having to pull up the man page for curl to check what each switch means.

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [reply .github/workflows/solutions_ci.yml:41]: Although note that since you are triggering a workflow, I don't think using the default `gh-token` will work, you will likely need to use `${{ secrets.ANSYS_SOLUTIONS_E2E_TESTS }}`

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [inline .github/workflows/solutions_ci.yml:3]: Pull requests against branches other than `main` are rare, but I want to make sure that is your intent here. Without a branch filter, this will apply to all PRs.

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [inline .github/workflows/solutions_ci.yml:1]: Nitpick - I know we generally use `ci.yml` for our workflows, but I have been pushing to change that to something more descriptive. For example, in solutions, we renamed `ci.yml` to `build-release.yml`.

My stance here stems from the fact that since we are in the `workflows` folder, we already know this is related to CI/CD. I compare this to stuff like `var myVar = ...` etc.

**Review by nick-marnik — COMMENTED** [review]
Just a few minor suggestions

File: .github/workflows/solutions_ci.yml
Comment (ccpansys) [reply .github/workflows/solutions_ci.yml:1]: Very fair

File: .github/workflows/solutions_ci.yml
Comment (ccpansys) [reply .github/workflows/solutions_ci.yml:3]: @nick-marnik Yes, it is intentional. Given that the tests will only run if changes to the solution templates have been made, I did not see the reason to also restrict branches. But please, let me know if you think there are reasons to add the filter!

File: .github/workflows/solutions_ci.yml
Comment (ccpansys) [reply .github/workflows/solutions_ci.yml:27]: @jacobevansgit I was not aware of that argument! Thanks for noticing, this is much better.

File: .github/workflows/solutions_ci.yml
Comment (ccpansys) [reply .github/workflows/solutions_ci.yml:17]: @nick-marnik I will, thank you for the suggestion.

**Comment by ccpansys** [comment]
@nick-marnik, @jacobevansgit: Thank you for your feedback!

File: .github/workflows/solutions_ci.yml
Comment (jacobevansgit) [inline .github/workflows/solutions_ci.yml:23]: Clean!

**Review by jacobevansgit — APPROVED** [review]

File: .github/workflows/solutions_ci.yml
Comment (nick-marnik) [reply .github/workflows/solutions_ci.yml:3]: The permissions on this repository are a bit unique, but for other private repos, I think including a `branches: main` filter makes sense because:
- branch protections are generally only configured for main
- if I have write access to a repo, I generally wouldn't use a PR to merge changes from `branchA` into `branchB` - I would simply do the merge myself and then push directly to `branchB` without a PR
- we generally don't use long-lived feature branches, so this would be an extremely rare operation
- if a e2e workflow is triggered for a feature branch, it may cause a failure and raise flags for investigation when it isn't necessary

It's not a hill I want to die on - the way it is won't hurt anything. I leave it up to you to determine if the filter is meaningful to add.

**Review by nick-marnik — APPROVED** [review]
NASA LGTM

Much cleaner - thanks for the quick turnaround on the changes! 👍

**Comment by nick-marnik** [comment]
@ccpansys This build triggered on a different PR and it looks like the secret may not exist: https://github.com/ansys/ansys-templates/actions/runs/6746000959/job/18338968194

Based on line 9, are you sure that `${{ secrets.ANSYS_SOLUTIONS_E2E_TESTS }}` exists on this repo? I don't have admin access to check myself 😢

**Comment by ccpansys** [comment]
@nick-marnik That is surprising. The workflow has run with that token before, so it must exist (or it did, at least). And I am afraid I don't have permissions to check, either. @iazehaf, can you help us figure this one out?

**Comment by nick-marnik** [comment]
I just checked and @ccpansys is correct, this build has run successfully in the past: https://github.com/ansys/ansys-templates/actions/workflows/solutions_ci.yml

I am just speculating, but I suspect this may be related to my permissions on this repo. Comparing the successful builds to my failed build, the difference is that my PR is based off of a branch from my personal fork of this repository. At one point in time, I correctly had write permissions for this repo and was able to push my branch directly. However, somewhere along the way of the public/private migrations of this repo, my permissions were broken and I was relegated to using a fork.

Paging @jacobevansgit for comment

PR in question: https://github.com/ansys/ansys-templates/pull/365

**Comment by jacobevansgit** [comment]
Sorry @nick-marnik  @ccpansys  I am also not admin of this org. Please reach out to Pyansys colleagues. 

However after looking at this PR again, I am wondering if we should have this _open_ repo signalling to start tests inside our internal org. The more appropriate way for this to scale would be to have the tests repo "listen" to this repository (`ansys-templates`) for any changes and then trigger if any are found. 

Updating the `ansys-templates` each time for a future 5-10-50 repositories that need testing after this repo has changes, is not appropriate IMO.

**Comment by nick-marnik** [comment]
> Sorry @nick-marnik @ccpansys I am also not admin of this org. Please reach out to Pyansys colleagues.
> 
> However after looking at this PR again, I am wondering if we should have this _open_ repo signalling to start tests inside our internal org. The more appropriate way for this to scale would be to have the tests repo "listen" to this repository (`ansys-templates`) for any changes and then trigger if any are found.
> 
> Updating the `ansys-templates` each time for a future 5-10-50 repositories that need testing after this repo has changes, is not appropriate IMO.

I don't disagree with your premise, but unfortunately I am not aware of a way to achieve this behavior. Currently, `workflow_run` triggers only work within the same repository, not between repos.

@jorgepiloto Do you have any idea what could be happening with the secret mentioned above?

**Comment by jorgepiloto** [comment]
Hi @nick-marnik. I can confirm that the secret ``ANSYS_SOLUTIONS_E2E_TESTS`` exists in this repository. It was updated two weeks ago. Have you authorized the TOKEN to operate with the Solution Applications org?

**Comment by nick-marnik** [comment]
> Hi @nick-marnik. I can confirm that the secret `ANSYS_SOLUTIONS_E2E_TESTS` exists in this repository. It was updated two weeks ago. Have you authorized the TOKEN to operate with the Solution Applications org?

I don't have access to see this, either, but I assume it is configured correctly because it works for some builds.

Are my permissions on this repo correct? I am curious if testing with a non-forked branch would yield different results.

**Pull request merged**
